src/test/java/com/iota/iri/service/TipsManagerTest.java

```diff
@@ -65,111 +75,269 @@ public static void tearDown() throws Exception {
-        transaction4 = new TransactionViewModel(getRandomTransactionWithTrunkAndBranch(transaction2.getHash(), transaction3.getHash()), getRandomTransactionHash());
+        transaction1 = new TransactionViewModel(getRandomTransactionWithTrunkAndBranch(
+                transaction.getHash(), transaction.getHash()), getRandomTransactionHash());
+        transaction2 = new TransactionViewModel(getRandomTransactionWithTrunkAndBranch(
+                transaction1.getHash(), transaction1.getHash()), getRandomTransactionHash());
+        transaction3 = new TransactionViewModel(getRandomTransactionWithTrunkAndBranch(
+                transaction2.getHash(), transaction2.getHash()), getRandomTransactionHash());
+        transaction4 = new TransactionViewModel(getRandomTransactionWithTrunkAndBranch(
+                transaction3.getHash(), transaction3.getHash()), getRandomTransactionHash());
         transaction.store(tangle);
         transaction1.store(tangle);
         transaction2.store(tangle);
         transaction3.store(tangle);
         transaction4.store(tangle);
-        Map<Hash, Set<Hash>> ratings = new HashMap<>();
-        tipsManager.updateHashRatings(transaction.getHash(), ratings, new HashSet<>());
-        Assert.assertEquals(ratings.get(transaction.getHash()).size(), 5);
-        Assert.assertEquals(ratings.get(transaction1.getHash()).size(),4);
-        Assert.assertEquals(ratings.get(transaction2.getHash()).size(), 3);
+
+        log.info(String.format("Linear ordered hashes from tip %.4s, %.4s, %.4s, %.4s, %.4s", transaction4.getHash(),
+                transaction3.getHash(), transaction2.getHash(), transaction1.getHash(), transaction.getHash()));
+
+        Map<Buffer, Integer> txToCw = tipsManager.calculateCumulativeWeight(new HashSet<>(),
+                transaction.getHash(), false, new HashSet<>());
+
+
+        Assert.assertEquals(String.format(TX_CUMULATIVE_WEIGHT_IS_NOT_AS_EXPECTED_FORMAT, 4),
+                1, txToCw.get(transaction4.getHash().getSubHash()).intValue());
+        Assert.assertEquals(String.format(TX_CUMULATIVE_WEIGHT_IS_NOT_AS_EXPECTED_FORMAT, 3),
+                2, txToCw.get(transaction3.getHash().getSubHash()).intValue());
+        Assert.assertEquals(String.format(TX_CUMULATIVE_WEIGHT_IS_NOT_AS_EXPECTED_FORMAT, 2),
+                3, txToCw.get(transaction2.getHash().getSubHash()).intValue());
+        Assert.assertEquals(String.format(TX_CUMULATIVE_WEIGHT_IS_NOT_AS_EXPECTED_FORMAT, 1),
+                4, txToCw.get(transaction1.getHash().getSubHash()).intValue());
+        Assert.assertEquals(String.format(TX_CUMULATIVE_WEIGHT_IS_NOT_AS_EXPECTED_FORMAT, 0),
+                5, txToCw.get(transaction.getHash().getSubHash()).intValue());
     }
 
     @Test
-    public void updateRatings2TestWorks() throws Exception {
-        TransactionViewModel transaction, transaction1, transaction2, transaction3, transaction4;
+    public void testCalculateCumulativeWeightAlon() throws Exception {
```

Comment:
Which Alon? :)
I think the name needs to say a bit more

Reply:
It is a unit test with a tangle @alon-e made up to understand the PR better. I have no idea what name to give it, honestly.

Any name you can recommend will be fine.

Reply:
Something to describe what it does.